Analytics/Classes/Internal/SEGUserDefaultsStorage.h

```diff
@@ -0,0 +1,20 @@
+//
+//  SEGUserDefaultsStorage.h
+//  Analytics
+//
+//  Created by Tony Xiao on 8/24/16.
+//  Copyright © 2016 Segment. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SEGStorage.h"
+
+@interface SEGUserDefaultsStorage : NSObject <SEGStorage>
+
+@property (nonatomic, strong, nullable) id<SEGCrypto> crypto;
```

Comment:
It seems a bit leaky that the the storage mechanism knows about encryption. Also causes fair bit of duplicated code b/w the two storage implementations to deal with crypto.

I think the encryption should happen before call the storage methods, and decryption after.

The usage could be like this:

```
NSData *encrypted = [crypto encrypt:data];
[storage setData:encrypted forKey:key];
```

```
NSData *encrypted = [storage dataForKey:key];
NSData *decrypted = [crypto decrypt:encrypted];
```